src/Futhark/Optimise/ArrayShortCircuiting/ArrayCoalescing.hs

```diff
@@ -1462,12 +1464,12 @@ genSSPointInfoSegOp
   scopetab
   (Pat [PatElem dst (_, MemArray dst_pt _ _ (ArrayIn dst_mem dst_ixf))])
   certs
-  (SegMap _ space _ kernel_body)
-    | (src, MemBlock _ shp src_mem src_ixf) : _ <-
+  (SegMap _ space _ kernel_body@KernelBody {kernelBodyResult = [Returns {}]})
```

Comment:
Yes, I think so. #2010 turned out to reveal multiple bugs in the short-circuiting code, and while the first fix (30406d4) did cause the the test in #2010 to pass (probably only because the input size was too small), it didn't actually address a more core issue, which is that we should never attempt to use the result of a non-Returns map as a short-circuiting point.